linera-execution/src/execution_state_actor.rs

```diff
@@ -311,12 +311,20 @@ where
 
             ReadBlobContent { blob_id, callback } => {
                 let blob = self.system.read_blob_content(blob_id).await?;
-                callback.respond(blob);
+                let is_new = !self.system.used_blobs.contains(&blob_id).await?;
+                if is_new {
+                    self.system.used_blobs.insert(&blob_id)?;
+                }
```

Comment:
As discussed - let's add this to a new method `read_blob`

Reply:
Or `use_blob`? `AssertBlobExists` doesn't really read it, just checks if it exists in storage

Reply:
Done. Although I had to make the transaction tracker an option: For system operations it's passed in and we have it in the execution state's methods, but for requests to the execution state actor we don't.

Changing this would be a major refactoring (which maybe we should consider at some point): Wrapping the execution state in an `ExecutionStateActor` that can hold additional data like the tracker (and possibly some other fields that are currently in the runtime).